lib/engine/game/g_1848/game.rb

```diff
@@ -435,6 +439,22 @@ def must_buy_train?(entity)
             !depot.depot_trains.empty? &&
              (self.class::MUST_BUY_TRAIN == :route && @graph.route_info(entity)&.dig(:route_train_purchase))
         end
+
+        def after_buy_company(player, company, _price)
+          # share_price = 100
+          # # NOTE: This should only ever be P6
+          abilities(company, :shares) do |ability|
+            ability.shares.each do |share|
+              if share.president
+                stock_market.set_par(share.corporation, stock_market.par_prices.find { |p| p.price == 100 })
+                share_pool.buy_shares(player, share, exchange: :free)
+                after_par(share.corporation)
```

Comment:
after_par(share.corporation) checks if it's incremental cap game (it's not)
 and does floating operations (in this game you need more than prez share to float), so irrelevant
 So I could remove this line, but saw it in other games, wanted to be consistent

Reply:
The code base changes quickly and after_par may grow functionality in the future that you'll need, so its best to leave it in there even if nothing happens for now.

Reply:
ok great, tx